javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -525,6 +525,31 @@ open class HandleManagerTestBase {
 
         handle.store(modified)
         assertThat(handle.fetchAll()).containsExactly(modified)
+    }   
+
+    @Test
+    open fun clientCanSetEntityId() = testRunner {
+        fakeTime.millis = 0
+        // Ask faketime to increment to test with changing timestamps.
+        fakeTime.autoincrement = 1
+        val id = "MyId"
+        val entity = TestParticle_Entities(text = "Hello", number = 1.0, entityId = id)
+        val handle = writeHandleManager.createCollectionHandle(entitySpec = TestParticle_Entities)
+        withContext(handle.dispatcher) {
+            handle.store(entity)
+            assertThat(handle.fetchAll()).containsExactly(entity)
+
+            // A different entity, with the same ID, should replace the first.
+            val entity2 = TestParticle_Entities(text = "New Hello", number = 1.1, entityId = id)
+            handle.store(entity2)
+            assertThat(handle.fetchAll()).containsExactly(entity2)
+            assertThat(entity2.creationTimestamp).isNotEqualTo(RawEntity.UNINITIALIZED_TIMESTAMP)
```

Comment:
could we explicitly test that the expirationTimestamp gets set to the minimum value, i.e. e2 overrode the `text` field, but not the timestamps?

Reply:
that's not what happens. we are substituting the entity entirely here in the storage proxy (crdt merge only happens in the storage level). Clarified in the test. if you think this is problem we can change the storage behavior, but not this.

Reply:
I'm pretty sure it's ok, but I'm confused about this and :
```
// Two different values, take minimum.
_data.creationTimestamp = minOf(_data.creationTimestamp, other.creationTimestamp)
```
in CrdtEntity.kt, could you, please, explain? feel free to merge, if we don't get a chance to chat today.